dna/src/commonMain/kotlin/predicate/PredicateFunctionContract.kt

```diff
@@ -0,0 +1,17 @@
+package com.github.trilobiitti.trilobite.dna.predicate
+
+import com.github.trilobiitti.trilobite.dna.data.DataType
+import kotlin.reflect.KClass
+
+
+interface PredicateFunctionContract {
+    // Throws if argument types are not acceptable
+    // Returns result type if they are
+    fun validateArgumentTypes(inputs: List<PredicateFunctionInput>): DataType<*>
+}
+
+
+class NotValidPredicateFunctionInputError(val availableFunctionInputClasses: List<KClass<*>> )
+    : Error("Not valid predicate function input. Allowed classes:" +
+        " ${availableFunctionInputClasses.map { it.simpleName }.joinToString(separator = " ,") }}." +
+        "Combination of these classes is not allowed")
```

Comment:
There is no rule disallowing combinations of literals and non-literals as inputs of a predicate function in general case. Also this exception class doesn't cover all possible scenarios of inputs invalidity, so let's just reuse `ValidationError` here instead of creating a new error class for now.